redfish-core/lib/license_service.hpp

```diff
@@ -0,0 +1,692 @@
+    const std::string& licenseType)
+{
+    if (licenseType == "com.ibm.License.Entry.LicenseEntry.Type.Purchased")
+    {
+        asyncResp->res.jsonValue["LicenseType"] = "Production";
+    }
+    else if (licenseType == "com.ibm.License.Entry.LicenseEntry.Type.Prototype")
+    {
+        asyncResp->res.jsonValue["LicenseType"] = "Prototype";
+    }
+    else if (licenseType == "com.ibm.License.Entry.LicenseEntry.Type.Trial")
+    {
+        asyncResp->res.jsonValue["LicenseType"] = "Trial";
+    }
+    else
+    {
+        // Any other values would be invalid
+        BMCWEB_LOG_ERROR("LicenseType value was not valid: {}", licenseType);
+        messages::internalError(asyncResp->res);
+    }
+}
+
+inline void translateAuthorizationTypeDbusToRedfish(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& authorizationType)
+{
+    if (authorizationType ==
+        "com.ibm.License.Entry.LicenseEntry.AuthorizationType.Unlimited")
+    {
+        asyncResp->res.jsonValue["AuthorizationScope"] = "Service";
+    }
+    else if (authorizationType ==
+             "com.ibm.License.Entry.LicenseEntry.AuthorizationType.Device")
+    {
+        asyncResp->res.jsonValue["AuthorizationScope"] = "Device";
+    }
+    else if (authorizationType ==
+             "com.ibm.License.Entry.LicenseEntry.AuthorizationType.Capacity")
+    {
+        asyncResp->res.jsonValue["AuthorizationScope"] = "Capacity";
+    }
+    else
+    {
+        // Any other values would be invalid
+        BMCWEB_LOG_ERROR("AuthorizationType value is not valid: {}",
+                         authorizationType);
+        messages::internalError(asyncResp->res);
+    }
+}
+
+inline void
+    getLicenseEntryById(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                        const std::string& licenseEntryID)
+{
+    crow::connections::systemBus->async_method_call(
+        [asyncResp, licenseEntryID](const boost::system::error_code ec,
+                                    GetManagedObjectsType& resp) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR("LicenseEntry resp_handler got error {}", ec);
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        std::string licenseEntryPath = "/xyz/openbmc_project/license/entry/" +
+                                       licenseEntryID;
+
+        for (const auto& objectPath : resp)
+        {
+            if (objectPath.first.str != licenseEntryPath)
+            {
+                continue;
+            }
+
+            uint64_t expirationTime = 0;
+            const uint32_t* deviceNumPtr = nullptr;
+            const std::string* serialNumPtr = nullptr;
+            const std::string* licenseNamePtr = nullptr;
+            const std::string* licenseTypePtr = nullptr;
+            const std::string* authorizationTypePtr = nullptr;
+            bool available = false;
+            bool state = false;
```

Comment:
The parse of the fields is better to be done via
```
sdbusplus::unpackPropertiesNoThrow()
```

Reply:
Consider in separate patch